lib/tasks/reset_password_email.rake

```diff
@@ -0,0 +1,68 @@
+# BigBlueButton open source conferencing system - http://www.bigbluebutton.org/.
+#
+# Copyright (c) 2022 BigBlueButton Inc. and by respective authors (see below).
+#
+# This program is free software; you can redistribute it and/or modify it under the
+# terms of the GNU Lesser General Public License as published by the Free Software
+# Foundation; either version 3.0 of the License, or (at your option) any later
+# version.
+#
+# Greenlight is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with Greenlight; if not, see <http://www.gnu.org/licenses/>.
+
+# frozen_string_literal: true
+
+require_relative 'task_helpers'
+
+namespace :migration do
+  batch_size = 500
+
+  desc 'Send a reset password email to users'
+  task :reset_password_email, %i[base_url provider] => :environment do |_task, args|
+    args.with_defaults(provider: 'greenlight')
+
+    root_url = "#{args[:base_url]}/"
+
+    if ENV['SMTP_SERVER'].blank?
+      err 'SMTP Server not set. Skipping sending reset password emails.'
+      exit 0
+    end
+
+    info 'Checking connection to SMTP Server...'
+    if ENV['SMTP_SERVER'].present?
+      begin
+        UserMailer.with(to: ENV.fetch('SMTP_SENDER_EMAIL', nil), subject: ENV.fetch('SMTP_SENDER_EMAIL', nil)).test_email.deliver_now
+      rescue StandardError => e
+        failed("Unable to connect to SMTP Server - #{e}")
+        exit 0
+      end
+    end
```

Comment:
I have added this block (copied from configuration.rake) to exit the rake tasks if the connection to the SMTP server fails.